lib/branch.test.js

```diff
@@ -2,35 +2,38 @@ const { isBranchNameValid } = require("./branch");
 
 describe("branch", () => {
   test("recognize valid branch names", () => {
-    expect(isBranchNameValid("core/foo")).toBe(true);
-    expect(isBranchNameValid("feature/foo")).toBe(true);
+    expect(isBranchNameValid("feat/foo")).toBe(true);
     expect(isBranchNameValid("fix/foo")).toBe(true);
-    expect(isBranchNameValid("hotfix/foo")).toBe(true);
-    expect(isBranchNameValid("asset/foo")).toBe(true);
-    expect(isBranchNameValid("rework/foo")).toBe(true);
-    expect(isBranchNameValid("rework/DriveFooterBar")).toBe(true);
-    expect(isBranchNameValid("documentation/foo")).toBe(true);
+    expect(isBranchNameValid("refactor/foo")).toBe(true);
+    expect(isBranchNameValid("docs/foo")).toBe(true);
+    expect(isBranchNameValid("chore/foo--bar")).toBe(true);
+    expect(isBranchNameValid("chore/foo--bar--z")).toBe(true);
+    expect(isBranchNameValid("feat/branch-name_from_linear")).toBe(true);
+    expect(isBranchNameValid("mobsuccessbot/foo")).toBe(true);
+    expect(isBranchNameValid("mobsuccessbot/foo@foo")).toBe(true);
+    expect(isBranchNameValid("mobsuccessbot/foo@foo/foo")).toBe(true);
     expect(isBranchNameValid("dependabot/foo")).toBe(true);
     expect(isBranchNameValid("dependabot/npm_and_yarn/axios-0.21.1")).toBe(
       true
     );
-    expect(isBranchNameValid("mobsuccessbot/foo")).toBe(true);
-    expect(isBranchNameValid("mobsuccessbot/foo@foo")).toBe(true);
-    expect(isBranchNameValid("mobsuccessbot/foo@foo/foo")).toBe(true);
-    expect(isBranchNameValid("foo/foo-bar")).toBe(false);
-    expect(isBranchNameValid("core/foo--bar")).toBe(true);
-    expect(isBranchNameValid("core/foo--bar--z")).toBe(true);
-    expect(isBranchNameValid("feature/branch-name_from_linear")).toBe(true);
   });
   test("recognize invalid branch names", () => {
-    expect(isBranchNameValid("core/npm_and_yarn/axios-0.21.1")).toBe(false);
-    expect(isBranchNameValid("core")).toBe(false);
-    expect(isBranchNameValid("core/")).toBe(false);
-    expect(isBranchNameValid("core/FOO")).toBe(false);
+    expect(isBranchNameValid("core/foo--bar")).toBe(false);
```

Comment:
test.each([
  "core/foo--bar--z",
  "foo/foo-bar",
  ...
])("The branch %s should be invalid", (givenBranch) => {
  expect(isBranchNameValid(givenBranch)).toBeFalsy();
});

Reply:
propre 👌 
[improve tests](https://github.com/mobsuccess-devops/github-actions-mobsuccess/pull/91/commits/37eacea32ea190030285d2211fc970eac90974c9)

Reply:
beaucoup mieux en effet